src/routes/posts/+page.server.ts

```diff
@@ -3,18 +3,24 @@ import type { Article } from '$lib/types';
 
 export const load: PageServerLoad = async ({ platform, url }) => {
 	const page = parseInt(url.searchParams.get('page') ?? '0');
+	const category = url.searchParams.get('category')
+		? parseInt(url.searchParams.get('category') as string)
+		: null;
+	const start = url.searchParams.get('start') ?? '2022-12-31';
+	const end = url.searchParams.get('end') ?? new Date().toISOString().split('T')[0];
```

Comment:
You may want to use something like a regex to sanitize these. Something like `\d{4}-\d{2}-\d{2}` to match the ISO 8601 format.